bindings/rust/evmc-vm/src/lib.rs

```diff
@@ -138,6 +169,74 @@ impl ExecutionResult {
     }
 }
 
+impl StepResult {
+    pub fn new(
+        step_status_code: StepStatusCode,
+        status_code: StatusCode,
+        revision: Revision,
+        pc: u64,
+        gas_left: i64,
+        gas_refund: i64,
+        output: Option<Vec<u8>>,
+        stack: Vec<Uint256>,
+        memory: Vec<u8>,
+        last_call_return_data: Option<Vec<u8>>,
+    ) -> Self {
+        Self {
+            step_status_code,
+            status_code,
+            revision,
+            pc,
+            gas_left,
+            gas_refund,
+            output,
+            stack,
+            memory,
+            last_call_return_data,
+        }
+    }
+
+    pub fn step_status_code(&self) -> StatusCode {
```

Comment:
I don't see the getters being used. And this makes me think if we need the constructor neither. The reason being: a large list of arguments are error-prone,  
easy to swap by accident. Struct constructor, on the other hand, requires field naming.  A POD struct with public members may be less code and less error-prone. 

After all, this is only a transient type that is going to live between return StepN call and evmc_step_result cast

Reply:
I completely agree. Making the struct fields private and having getters for every field although the struct has no internal invariants is somewhat pointless. However I wanted to keep it as similar to `ExecutionResult` as possible, and there they do the same thing.

Reply:
Ok, coherence in style is a good point as well.  maybe we drop dead code (getters)